src/components/AddressInput.tsx

```diff
@@ -1,32 +1,54 @@
 import { createEffect, on } from "solid-js";
 
-import { RBTC } from "../consts";
+import { LN, RBTC } from "../consts";
 import { useCreateContext } from "../context/Create";
 import { useGlobalContext } from "../context/Global";
-import { decodeAddress } from "../utils/compat";
-import { extractAddress } from "../utils/invoice";
+import { decodeAddress, probeAddress } from "../utils/compat";
+import { extractAddress, extractInvoice, isInvoice } from "../utils/invoice";
 
 const AddressInput = () => {
     let inputRef: HTMLInputElement;
 
-    const { t } = useGlobalContext();
+    const { t, notify } = useGlobalContext();
     const {
         asset,
         assetReceive,
         reverse,
         amountValid,
         onchainAddress,
         setAddressValid,
+        setAssetReceive,
+        setAssetSend,
         setOnchainAddress,
         setButtonLabel,
+        setInvoice,
     } = useCreateContext();
 
     const validateAddress = (input: HTMLInputElement) => {
         const inputValue = input.value.trim();
         const address = extractAddress(inputValue);
+        const invoice = extractInvoice(inputValue);
+
+        // auto switch direction based on invoice
+        if (isInvoice(invoice)) {
+            setAssetReceive(LN);
+            setAssetSend(asset());
+            setInvoice(invoice);
+            notify("success", t("switch_paste"));
```

Comment:
Not a fan of throwing a notification for every action you do

Reply:
i like it, and i talked with @kilrau about it.